zipline/utils/calendars/trading_calendar.py

```diff
@@ -0,0 +1,843 @@
+
+    def next_session_label(self, session_label):
+        """
+        Given a session label, returns the label of the next session.
+
+        Parameters
+        ----------
+        session_label: pd.Timestamp
+            A session whose next session is desired.
+
+        Returns
+        -------
+        pd.Timestamp
+            The next session label (midnight UTC).
+
+        Notes
+        -----
+        Raises ValueError if the given session is the last session in this
+        calendar.
+        """
+        idx = self.schedule.index.get_loc(session_label)
+        try:
+            return self.schedule.index[idx + 1]
+        except IndexError:
+            if idx == len(self.schedule.index) - 1:
+                raise ValueError("There is no next session as this is the end"
+                                 "of the exchange calendar.")
+
+    def previous_session_label(self, session_label):
+        """
+        Given a session label, returns the label of the previous session.
+
+        Parameters
+        ----------
+        session_label: pd.Timestamp
+            A session whose previous session is desired.
+
+        Returns
+        -------
+        pd.Timestamp
+            The previous session label (midnight UTC).
+
+        Notes
+        -----
+        Raises ValueError if the given session is the first session in this
+        calendar.
+        """
+        idx = self.schedule.index.get_loc(session_label)
+        if idx == 0:
+            raise ValueError("There is no previous session as this is the"
+                             "beginning of the exchange calendar.")
+
+        return self.schedule.index[idx - 1]
+
+    def minutes_for_session(self, session_label):
+        """
+        Given a session label, return the minutes for that session.
+
+        Parameters
+        ----------
+        session_label: pd.Timestamp (midnight UTC)
+            A session label whose session's minutes are desired.
+
+        Returns
+        -------
+        pd.DateTimeIndex
+            All the minutes for the given session.
+        """
+        data = self.schedule.loc[session_label]
+        return self.all_minutes[
+            self.all_minutes.slice_indexer(
+                data.market_open,
+                data.market_close
+            )
+        ]
+
+    def minutes_window(self, start_dt, count):
+        try:
+            start_idx = self.all_minutes.get_loc(start_dt)
+        except KeyError:
+            # if this is not a market minute, go to the previous session's
+            # close
+            previous_session = self.minute_to_session_label(
+                start_dt, direction="previous"
+            )
+
+            previous_close = self.open_and_close_for_session(
+                previous_session
+            )[1]
+
+            start_idx = self.all_minutes.get_loc(previous_close)
+
+        end_idx = start_idx + count
+
+        if start_idx > end_idx:
```

Comment:
if the user passed in a negative value for the `count` parameter.  ie, start at `start_dt` and go `count` minutes backwards in time.